From d5492ae7a0fa4e2c19b1d091ad28133fed53d8e2 Mon Sep 17 00:00:00 2001
From: Thomas Devoogdt <thomas.devoogdt@barco.com>
Date: Tue, 25 Jul 2023 09:10:41 +0200
Subject: [PATCH] lib: librdkafka: CMakeLists.txt: allow compilation without
 CXX support #7741

e.g. fluent-bit is a c-only library, so allow compilation without cxx

Upstream: https://github.com/confluentinc/librdkafka/pull/4366
Upstream: https://github.com/fluent/fluent-bit/pull/7765
Signed-off-by: Thomas Devoogdt <thomas.devoogdt@barco.com>
---
 lib/librdkafka-2.1.0/CMakeLists.txt          | 18 ++++++++++--
 lib/librdkafka-2.1.0/examples/CMakeLists.txt | 30 +++++++++++++-------
 2 files changed, 35 insertions(+), 13 deletions(-)

diff --git a/lib/librdkafka-2.1.0/CMakeLists.txt b/lib/librdkafka-2.1.0/CMakeLists.txt
index 7f3dd0fc6..c1ee93062 100644
--- a/lib/librdkafka-2.1.0/CMakeLists.txt
+++ b/lib/librdkafka-2.1.0/CMakeLists.txt
@@ -3,7 +3,12 @@ cmake_minimum_required(VERSION 3.2)
 include("packaging/cmake/parseversion.cmake")
 parseversion("src/rdkafka.h")
 
-project(RdKafka VERSION ${RDKAFKA_VERSION})
+project(RdKafka VERSION ${RDKAFKA_VERSION} LANGUAGES C)
+
+option(RDKAFKA_BUILD_CXX "Enable CXX support" ON)
+if(RDKAFKA_BUILD_CXX)
+  enable_language(CXX)
+endif()
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/packaging/cmake/Modules/")
 
@@ -181,7 +186,11 @@ endif()
 
 option(RDKAFKA_BUILD_STATIC "Build static rdkafka library" OFF)
 option(RDKAFKA_BUILD_EXAMPLES "Build examples" ON)
-option(RDKAFKA_BUILD_TESTS "Build tests" ON)
+
+if(RDKAFKA_BUILD_CXX)
+    option(RDKAFKA_BUILD_TESTS "Build tests" ON)
+endif()
+
 if(WIN32)
     option(WITHOUT_WIN32_CONFIG "Avoid including win32_config.h on cmake builds" ON)
 endif(WIN32)
@@ -279,7 +288,10 @@ install(
 )
 
 add_subdirectory(src)
-add_subdirectory(src-cpp)
+
+if(RDKAFKA_BUILD_CXX)
+  add_subdirectory(src-cpp)
+endif()
 
 if(RDKAFKA_BUILD_EXAMPLES)
   add_subdirectory(examples)
diff --git a/lib/librdkafka-2.1.0/examples/CMakeLists.txt b/lib/librdkafka-2.1.0/examples/CMakeLists.txt
index bbbb89ad9..51b8b04be 100644
--- a/lib/librdkafka-2.1.0/examples/CMakeLists.txt
+++ b/lib/librdkafka-2.1.0/examples/CMakeLists.txt
@@ -5,8 +5,10 @@ endif(WIN32)
 add_executable(producer producer.c ${win32_sources})
 target_link_libraries(producer PUBLIC rdkafka)
 
-add_executable(producer_cpp producer.cpp ${win32_sources})
-target_link_libraries(producer_cpp PUBLIC rdkafka++)
+if(RDKAFKA_BUILD_CXX)
+    add_executable(producer_cpp producer.cpp ${win32_sources})
+    target_link_libraries(producer_cpp PUBLIC rdkafka++)
+endif()
 
 add_executable(consumer consumer.c ${win32_sources})
 target_link_libraries(consumer PUBLIC rdkafka)
@@ -14,14 +16,20 @@ target_link_libraries(consumer PUBLIC rdkafka)
 add_executable(rdkafka_performance rdkafka_performance.c ${win32_sources})
 target_link_libraries(rdkafka_performance PUBLIC rdkafka)
 
-add_executable(rdkafka_example_cpp rdkafka_example.cpp ${win32_sources})
-target_link_libraries(rdkafka_example_cpp PUBLIC rdkafka++)
+if(RDKAFKA_BUILD_CXX)
+    add_executable(rdkafka_example_cpp rdkafka_example.cpp ${win32_sources})
+    target_link_libraries(rdkafka_example_cpp PUBLIC rdkafka++)
+endif()
 
-add_executable(rdkafka_complex_consumer_example_cpp rdkafka_complex_consumer_example.cpp ${win32_sources})
-target_link_libraries(rdkafka_complex_consumer_example_cpp PUBLIC rdkafka++)
+if(RDKAFKA_BUILD_CXX)
+    add_executable(rdkafka_complex_consumer_example_cpp rdkafka_complex_consumer_example.cpp ${win32_sources})
+    target_link_libraries(rdkafka_complex_consumer_example_cpp PUBLIC rdkafka++)
+endif()
 
-add_executable(openssl_engine_example_cpp openssl_engine_example.cpp ${win32_sources})
-target_link_libraries(openssl_engine_example_cpp PUBLIC rdkafka++)
+if(RDKAFKA_BUILD_CXX)
+    add_executable(openssl_engine_example_cpp openssl_engine_example.cpp ${win32_sources})
+    target_link_libraries(openssl_engine_example_cpp PUBLIC rdkafka++)
+endif()
 
 add_executable(misc misc.c ${win32_sources})
 target_link_libraries(misc PUBLIC rdkafka)
@@ -35,6 +43,8 @@ if(NOT WIN32)
     add_executable(rdkafka_complex_consumer_example rdkafka_complex_consumer_example.c)
     target_link_libraries(rdkafka_complex_consumer_example PUBLIC rdkafka)
 
-    add_executable(kafkatest_verifiable_client kafkatest_verifiable_client.cpp)
-    target_link_libraries(kafkatest_verifiable_client PUBLIC rdkafka++)
+    if(RDKAFKA_BUILD_CXX)
+        add_executable(kafkatest_verifiable_client kafkatest_verifiable_client.cpp)
+        target_link_libraries(kafkatest_verifiable_client PUBLIC rdkafka++)
+    endif()
 endif(NOT WIN32)
-- 
2.34.1

